### PR TITLE
created initial about.md file for feature extraction pipeline

### PR DESCRIPTION
**Pull request by shloksah**
Added Use cases- Sentence classification & Inference
Will adding more as I research on the topic. Thanks

### CODE REVIEW

**Comment by shloksah** [comment]
partially fixes #355

File: tasks/src/feature-extraction/about.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+print(features_df.shape)
+
+# Output:
+# (4, 768)
+
+# For a list of inputs
+input_list = ['Hello world','How are you?']
+features_list = feature_extraction(input_list)
+
+for index, features in enumerate(features_list):
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:35]: Can you somehow explain to a person that doesn't know about this task what do numbers (output) mean here? 🙂 
You can also mention about how people can use these pipelines to train classifiers on top (without fine-tuning) as I feel like it's one of the main points.

File: tasks/src/feature-extraction/about.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    features = np.squeeze(features,axis = 0)
+    features_df = pd.DataFrame(features)
+    print(f'Shape of input: {input_list[index]} = {features_df.shape}')
+
+# Output:
+# Shape of input: Hello world = (4, 768)
+# Shape of input: How are you? = (6, 768)
+```
+## Useful Resources
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:44]: Can you copy and paste this template: https://github.com/huggingface/hub-docs/blob/main/tasks/src/placeholder/about.md and make your changes on top of it? It helps other contributors, us to keep encourage folks and it means less merge conflicts later on 🙂

File: tasks/src/feature-extraction/about.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+## Use cases
+
+### Sentence classification
+
+For sentence classification, we’re only only interested in BERT’s output for the [CLS] token, so we get the sentence embedding using
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:5]: Can you explain this simpler? You can see other task pages and the contribution guidelines for the technical writing we look for 🙂

**Review by merveenoyan — COMMENTED** [review]
Hello there 🙂 Left a very general feedback, thanks a lot for working on this!!!

**Comment by shloksah** [comment]
Hi @merveenoyan , I added a code snippet to train a Random Forest on embeddings. Just a little confused if I should place it in Task variants or Use cases. Also, I'm working on improving the technical writing you mentioned. Thanks

File: tasks/src/feature-extraction/about.md (hunk 2 above)
Comment (merveenoyan) [reply tasks/src/feature-extraction/about.md:122]: Can you address this? 🙂

File: tasks/src/feature-extraction/about.md [hunk 4]
```diff
@@ -42,3 +120,7 @@ for index, features in enumerate(features_list):
 # Shape of input: How are you? = (6, 768)
 ```
 ## Useful Resources
+
+https://huggingface.co/docs/transformers/v4.24.0/en/main_classes/pipelines#transformers.FeatureExtractionPipeline
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:124]: Can you write explanations and put them in markdown links? 
e.g. `[your-explanation-here](https://huggingface.co/docs/transformers/v4.24.0/en/main_classes/pipelines#transformers.FeatureExtractionPipeline
Write)`

File: tasks/src/feature-extraction/about.md [hunk 5]
```diff
@@ -1,15 +1,93 @@
...
+```
+
+Taking top 100 rows from Newsgroup classification data from Sklearn
+
+```python
+dataset =  datasets.fetch_20newsgroups()
+X = dataset.data[:100]
+y = dataset.target[:100]
+```
+This function creates a sparse embedding for each token present in the sentence. Please remember to use the same tokenizer that you would be using while calling the feature-extraction pipeline. As the default BERT tokenizer used in the pipeline is bert-base-cased so we use the same for initializing the dataframe with number of columns = vocabulary size of tokenizer. We are using dataframe within the function for ease of understanding and it returns an np.array of size (n_samples,vocab_size)
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:35]:
```suggestion
This function creates a sparse embedding for each token present in the sentence. Please remember to use the same tokenizer that you would be using while calling the `feature-extraction` pipeline. As the default, BERT tokenizer used in the pipeline is bert-base-cased, so we use the same for initializing the dataframe with number of columns equal to vocabulary size of the tokenizer. We are using a dataframe within the function for ease of understanding, and it returns an `np.array` of size `(n_samples,vocab_size)`.
```

File: tasks/src/feature-extraction/about.md [hunk 6]
```diff
@@ -1,15 +1,93 @@
...
+```python
+model = RandomForestClassifier(random_state=42)
+model.fit(X_train,y_train)
+y_pred = model.predict(X_test)
+print(f'Test Accuracy is {accuracy_score(y_test,y_pred)*100} %')
+
+# Output:
+# Test Accuracy is 95.0 %
+```
+This exercise shows the potential of using embeddings as an input to a classifier. For imbalanced dataset, we can use performance metrics like F1 and ROC_AUC.
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:81]:
```suggestion
This exercise shows the potential of using embeddings as input to a classifier. For an imbalanced dataset, we can use performance metrics like F1-score and ROC AUC.
```

File: tasks/src/feature-extraction/about.md [hunk 7]
```diff
@@ -1,15 +1,93 @@
...
+
+
+# Task Variants
 
 ## Inference
 
-You can infer with text data with the 🤗 Transformers library using the `feature-extraction` pipeline. This pipeline takes a string or a list of string input and generates features for each input. All models may be used for this pipeline (see a list of all models on https://huggingface.co/models).
+This pipeline extracts the hidden states from the base transformer, which can be used as features in downstream tasks. You can infer the text data with the 🤗 Transformers library using the `feature-extraction` pipeline. This pipeline takes a string or a list of string input and generates features for each input. All models may be used for this pipeline (see a list of all models on https://huggingface.co/models).
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:88]: Can you try to explain what hidden state is at the beginning? Try to read the text as a beginner who doesn't know much about the task or ML 🙂

File: tasks/src/feature-extraction/about.md [hunk 8]
```diff
@@ -1,15 +1,93 @@
...
+
+The output generated for a single input is a shape of dataframe which contains 768 embeddings for each token. See the use-case above where we calculate mean of these embeddings and use them as input to a classifier.
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:90]:
```suggestion
The output generated for a single input is a shape of a dataframe, which contains 768 embeddings for each token. See the use-case above, where we calculate mean of these embeddings and use them as input to a classifier.
```

**Review by merveenoyan — COMMENTED** [review]
Left a very general feedback, thanks a lot! 🙂❤️

File: tasks/src/feature-extraction/about.md (hunk 2 above)
Comment (shloksah) [reply tasks/src/feature-extraction/about.md:122]: Apologies but I'm a bit confused - I copied the template and using it already. PFB the markdown headings that I copied from the given link 🙂

#Use Cases
#Task Variants
## Inference
## Useful Resources

File: tasks/src/feature-extraction/about.md (hunk 7 above)
Comment (merveenoyan) [reply tasks/src/feature-extraction/about.md:88]: Can you please address this comment? :')

File: tasks/src/feature-extraction/about.md (hunk 4 above)
Comment (merveenoyan) [reply tasks/src/feature-extraction/about.md:124]: Can you please address this? 🙂

File: tasks/src/feature-extraction/about.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
+# Use cases
+
+## Sentence classification
+
+For sentence classification, we are interested in BERT’s hidden layer output (embeddings), so we get the sentence embedding using
+`feature-extraction` pipeline which contain contextual knowledge of English language that can be used as input for Tree based classifiers like Random Forest, XGBoost, CatBoost.
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:6]:
```suggestion
`feature-extraction` pipeline which contain contextual knowledge of English language that can be used as input for Tree based classifiers like Random Forest, XGBoost, CatBoost. You can check out [this tutorial](https://towardsdatascience.com/feature-extraction-with-bert-for-text-classification-533dde44dc2f) for more in depth explanation. 
```

Comment (merveenoyan) [reply tasks/src/feature-extraction/about.md:6]: also I feel like it's any classifier really, so I'd rather avoid giving specifics and let people pick what they want.

**Review by merveenoyan — COMMENTED** [review]
Left readability nits 🙂

File: tasks/src/feature-extraction/about.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+# Task Variants
+
+## Inference
+
+This pipeline extracts the hidden states from the base transformer, which can be used as features in downstream tasks. You can infer the text data with the 🤗 Transformers library using the `feature-extraction` pipeline. This pipeline takes a string or a list of string input and generates features for each input. All models may be used for this pipeline (see a list of all models on https://huggingface.co/models).
+
+The output generated for a single input is a shape of a dataframe, which contains 768 embeddings for each token. See the use-case above, where we calculate mean of these embeddings and use them as input to a classifier.
+
+```python
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:92]: In this part it's best if we only show how to infer and use this pipeline for semantic similarity, does it sound good? Given you've put a lot of effort into this code, maybe we could simply mention this as another way to use this pipeline. We want to enable readability for beginner audience. Thanks a lot for your patience!

File: tasks/src/feature-extraction/about.md [hunk 11]
```diff
@@ -1,19 +1,21 @@
 # Use cases
 
+You can contribute this area with common use cases of the task!
+
 ## Sentence classification
 
-For sentence classification, we are interested in BERT’s hidden layer output (embeddings), so we get the sentence embedding using
-`feature-extraction` pipeline which contain contextual knowledge of English language that can be used as input for Tree based classifiers like Random Forest, XGBoost, CatBoost. 
-(see https://towardsdatascience.com/feature-extraction-with-bert-for-text-classification-533dde44dc2f).
+For sentence classification, we are interested in BERT’s hidden layer output (embeddings), so we get the sentence embeddings using
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:7]: Can you start this paragraph with some information on how feature extraction pipeline can be used and relates to text classification? Try to put yourself in a beginner's shoes 🙂

File: tasks/src/feature-extraction/about.md [hunk 12]
```diff
@@ -1,19 +1,21 @@
...
+`feature-extraction` pipeline which contain contextual knowledge of English language that can be used as input for any classifier. You can check out [this-tutorial](https://towardsdatascience.com/feature-extraction-with-bert-for-text-classification-533dde44dc2f) for more in-depth explanation.
```
Comment (merveenoyan) [inline tasks/src/feature-extraction/about.md:8]: Can you make this sentence a little shorter for easier read?

File: tasks/src/feature-extraction/about.md (hunk 11 above)
Comment (merveenoyan) [reply tasks/src/feature-extraction/about.md:7]: Also if it's okay I feel like this section (sentence classification use case) is a bit intermediate level info compressed here, it's very nicely done and very relevant. Is it okay if we make this use case a nicely done colab instead and put it to useful resources section? I will share it separately too!

**Review by merveenoyan — COMMENTED** [review]
I really like this, thanks a lot! 🙂🤗

**Comment by merveenoyan** [comment]
@shloksah do you need help on this? 🙂

**Comment by shloksah** [comment]
> @shloksah do you need help on this? 🙂

Hi @merveenoyan , I'm working on the suggestions sorry for a bit delay - I have my end semester exams this week. Thanks 😄…

**Comment by merveenoyan** [comment]
@shloksah no worries, best of luck!!! (I'm also a master's student 😂)

**Comment by merveenoyan** [comment]
@shloksah we deployed feature extraction here so there's a merge conflict, would you like to make additions on top of it instead? https://huggingface.co/tasks/feature-extraction

**Comment by merveenoyan** [comment]
Hey @shloksah 👋 
We have moved the codes for Tasks project to [huggingface.js repository](https://github.com/huggingface/huggingface.js) so if you feel like continuing this pull request you can move your work over there. Many thanks! 🤗

**Pull request closed without merging**